src/TorchSharp/Tensor/Tensor.cs

```diff
@@ -5142,6 +5142,64 @@ public IList<Tensor> nonzero_as_list()
                 return t.chunk(t.shape[1], dim: 1);
             }
 
+            [DllImport("LibTorchSharp")]
+            extern static IntPtr THSTensor_roll(IntPtr tensor, IntPtr shifts, int shLength, IntPtr dims, long dimLength);
+
+            /// <summary>
+            /// Roll the tensor along the given dimension(s).
+            /// Elements that are shifted beyond the last position are re-introduced at the first position.
+            /// If a dimension is not specified, the tensor will be flattened before rolling and then restored to the original shape.
+            /// </summary>
+            public Tensor roll(long shifts, long? dims = null)
+            {
+                if (dims.HasValue) {
+                    return roll(new long[] { shifts }, new long[] { dims.Value });
+                }
+                else {
+                    return roll(new long[] { shifts }, null);
+                }
+            }
+
+            /// <summary>
+            /// Roll the tensor along the given dimension(s).
+            /// Elements that are shifted beyond the last position are re-introduced at the first position.
+            /// If a dimension is not specified, the tensor will be flattened before rolling and then restored to the original shape.
+            /// </summary>
+            public Tensor roll((long,long) shifts, (long,long) dims)
+            {
+                return roll(new long[] { shifts.Item1, shifts.Item2 }, new long[] { dims.Item1, dims.Item2 });
+            }
+
+            /// <summary>
+            /// Roll the tensor along the given dimension(s).
+            /// Elements that are shifted beyond the last position are re-introduced at the first position.
+            /// If a dimension is not specified, the tensor will be flattened before rolling and then restored to the original shape.
+            /// </summary>
+            public Tensor roll((long, long, long) shifts, (long, long, long) dims)
+            {
+                return roll(new long[] { shifts.Item1, shifts.Item2, shifts.Item3 }, new long[] { dims.Item1, dims.Item2, dims.Item3 });
+            }
+
+            /// <summary>
+            /// Roll the tensor along the given dimension(s).
+            /// Elements that are shifted beyond the last position are re-introduced at the first position.
+            /// If a dimension is not specified, the tensor will be flattened before rolling and then restored to the original shape.
+            /// </summary>
+            public Tensor roll(IEnumerable<long> shifts, IEnumerable<long>? dims = null)
```

Comment:
>IEnumerable [](http://example.com/codeflow?start=31&length=11)

would it make sense more if you make this method takes either ReadOnlySpan<long> which will avoid the array allocation in the callsite and also will avoid calling ToArray here too.